firecrown/connector/numcosmo/numcosmo.py

```diff
@@ -193,9 +213,8 @@ def set_params_from_numcosmo(
             case _:
                 raise ValueError(f"NumCosmo object {type(hi_cosmo)} not supported.")
 
-        A_s, sigma8 = helpers.get_amplitude_parameters(
-            ccl_factory, self._p_ml, hi_cosmo
-        )
+        p_ml = self._p.linear if self._p is not None else None
```

Comment:
The following can be simplified by using `self.p_ml` instead of calculating `p_ml` as done above.